File: php/src/ICredentialsProvider.php
<?php

namespace Alibabacloud\CredentialsAPI;

use Alibabacloud\CredentialsAPI\ICredentials;

interface ICredentialsProvider {

    /**
     * @return ICredentials
     */
    public function getCredentials();

    /**
     * @return string
     */
    public function getProviderName();
}

File: java/src/main/java/com/aliyun/credentials/api/ICredentialsProvider.java
package com.aliyun.credentials.api;

import java.util.concurrent.CompletableFuture;

public interface ICredentialsProvider extends AutoCloseable {
    ICredentials getCredentials();

    default CompletableFuture<ICredentials> getCredentialsAsync() {
        return null;
    }

    String getProviderName();
}
